easybuild/easyblocks/o/opencv.py

```diff
@@ -164,8 +164,17 @@ def configure_step(self):
         # configure optimisation for CPU architecture
         # see https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options
         if self.toolchain.options.get('optarch') and 'CPU_BASELINE' not in self.cfg['configopts']:
+            optimal_arch_option = self.toolchain.COMPILER_OPTIMAL_ARCHITECTURE_OPTION.get(
+                (self.toolchain.arch, self.toolchain.cpu_family), '')
             optarch = build_option('optarch')
-            if not optarch:
+            if (
+                not optarch
+                or (isinstance(optarch, str) and optimal_arch_option in optarch)
+                or (isinstance(optarch, dict) and (
+                    (get_software_root('GCC') and optimal_arch_option in optarch.get('GCC', ''))
+                    or (get_software_root('intel-compilers') and optimal_arch_option in optarch.get('Intel', ''))
```

Comment:
for `intel` < `2021a` we would need to look for `iccifort` instead of `intel-compilers`?

Reply:
done in https://github.com/easybuilders/easybuild-easyblocks/pull/2954/commits/eb0ee642abc0ed58e4113373b44844aa7fcd1c4f